ambry-api/src/main/java/com.github.ambry/network/BoundedNettyByteBufReceive.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A netty {@link ByteBuf} version of Receive to buffer the incoming request or response.
+ */
+public class BoundedNettyByteBufReceive implements BoundedReceive<ByteBuf> {
```

Comment:
`BoundedByteBufReceive` is sufficient.

Reply:
I prefer adding Netty in the middle of the name, since the counterpart's name is BoundedByteBufferReceive, which is too close to BoundedByteBufReceive. Adding Netty in the middle of it, would better serve intellij as well as human eyes.